### PR TITLE
Added default user-agent (completing PR #314)

### PR DESCRIPTION
**Pull request by dstpierre**
I've changed the `version` package started in PR #314 to use a `Version` struct and a `Full` function that returns the current full version.

As requested in the PR, I've added the default user-agent in the `cmd` package options for the `run` command.

I've removed `Version` from `cmd/root.go` and refactored to use the new `version` package introduced in the #314 changes.

P.S. I wasn't sure in which branch to merge this PR, I picked the original PR feature branch, let me know if that would make more sense in the master branch or something.

Thanks

### CODE REVIEW

**Comment by liclac** [comment]
Please rebase against master!

File: version/version.go [hunk 1]
```diff
@@ -1,4 +1,42 @@
...
+	Major uint
+	Minor uint
+	Patch uint
+}
+
+// Current represents the current version and can be shared across packages
+var Current Version
+
+func init() {
+	Current = Version{Major: 0, Minor: 18, Patch: 2}
```
Comment (liclac) [inline version/version.go]: No need to do this in an init, this is valid:

```go
var Current = Version{Major: 0, Minor: 18, Patch: 2}
```

Or even this, if you want:

```go
var Current = Version{0, 18, 2}
```

File: version/version.go [hunk 2]
```diff
@@ -1,4 +1,42 @@
...
+}
+
+// Full returns the full semantic version as a string major.minor.patch
+func Full() string {
+	return fmt.Sprintf("%d.%d.%d", Current.Major, Current.Minor, Current.Patch)
```
Comment (liclac) [inline version/version.go]: I feel like this should be cached, or even generated at init time.

File: version/version.go (hunk 1 above)
Comment (dstpierre) [reply version/version.go]: yes, good point

File: version/version.go (hunk 2 above)
Comment (dstpierre) [reply version/version.go]: I was saying that to myself as well when doing the changes. Would you have a suggestion for this? From what I can see when you are doing the release this version should be modified and there's also the git tag.

I'll push the rebase and the removal of the init in the meantime.

**Comment by liclac** [comment]
Sorry, by rebase, I mean please make this PR against master. Right now the diff is utterly impossible to read because it's merging 300 commits from master into an outdated feature branch… ^^;;

**Comment by dstpierre** [comment]
@liclac oh my bad. I was not certain at all what to do with this PR since it was based on a topic branch that I did not had access to originally.

Thanks, hopefully that's a bit cleaner now.

File: cmd/run.go [hunk 3]
```diff
@@ -67,27 +68,27 @@ var runCmd = &cobra.Command{
 	Use:   "run",
 	Short: "Start a load test",
 	Long: `Start a load test.
-
-This also exposes a REST API to interact with it. Various k6 subcommands offer
-a commandline interface for interacting with it.`,
+
```
Comment (liclac) [inline cmd/run.go]: Please make another PR if you want to change these indents.

File: version/version.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package version
+
+import "fmt"
+
+// Version represents a sember - k6 uses semantic versioning (http://semver.org/)
+type Version struct {
```
Comment (liclac) [inline version/version.go]: I'm not entirely sold on the need for this, rather than just a string constant?
